Marlin/src/HAL/HAL_STM32/timers.cpp

```diff
@@ -36,58 +36,52 @@
 // Private Variables
 // ------------------------
 
-stm32_timer_t TimerHandle[NUM_HARDWARE_TIMERS];
+HardwareTimer *TimerHandle[NUM_HARDWARE_TIMERS];
+bool TimerEnabled[NUM_HARDWARE_TIMERS];
 
 // ------------------------
 // Public functions
 // ------------------------
 
 bool timers_initialized[NUM_HARDWARE_TIMERS] = { false };
 
+//frequency is in Hertz
 void HAL_timer_start(const uint8_t timer_num, const uint32_t frequency) {
```

Comment:
According to the method name it should also start the timer, not only initialize it.
Can anyone confirm this?

Reply:
According to http://www.awesomeband.nz/kieren/Arduino_Core_STM32/blob/master/cores/arduino/stm32/timer.c#L400 `TimerHandleInit` function starts the timer (see `HAL_TIM_Base_Start_IT` call).

Reply:
yes, you are right, I saw that too yesterday, that's why i made the next commit but now, I'm seeing that in stepper.cpp and temperature.cpp they "enable the interrupt" after starting the respective timer. they mean to start receiving interrupts....which is what "HardwareTimer.resume()" does.....so,even if in my last commit I automatically do that I should not. (okay calling start twice doesn't do any damage but still I shouldn't)